src/pages/admin/create/questions.tsx

```diff
@@ -0,0 +1,137 @@
+import { AddIcon } from '@chakra-ui/icons';
+import { Flex, Heading, Select, Text } from '@chakra-ui/react';
+import { ChangeEvent, useEffect, useState } from 'react';
+import useCreatorStorage, {
+  StoreQuestion,
+  StoreRound,
+} from '../../../hooks/use-creator-storage';
+import { CreatorModeQuestion } from '../../../../types';
+import SubHeader from '../../../components/headers/sub-header';
+import CreatorQuestion from '../../../components/creator-question';
+import SecondaryButton from '../../../components/buttons/secondary-button';
+import { AdminCreatorWrapper } from '../../../components/wrappers/admin-creator-wrapper';
+import RouteNavigation from '../../../components/route-navigation';
+import { ADMIN_CREATE_ROUTE_LIST } from '../../../../constants';
+
+const QUESTION_PRESET = { content: '', answer: '' };
+
+const Questions = () => {
+  const { initialData, setData } = useCreatorStorage();
+
+  const [rounds, setRounds] = useState<StoreRound[]>([]);
+
+  // @FIXME This not ideal to do it by index but we dont have ids yet
+  const [selectedRoundIndex, setSelectedRoundIndex] = useState(-1);
+
+  const [questions, setQuestions] = useState<StoreQuestion[][]>();
+
+  useEffect(() => {
+    const rounds = initialData.rounds || [];
+    setRounds(rounds);
+    setQuestions(rounds.map((round) => round.questions ?? []));
+  }, []);
+
+  const handleSelectRound = (e: ChangeEvent<HTMLSelectElement>) => {
+    const roundIndex = e.target.value === '' ? -1 : Number(e.target.value);
+
+    setSelectedRoundIndex(roundIndex);
+
+    if (roundIndex > -1) {
+      setQuestions((currQuestions) => {
+        if (currQuestions?.[roundIndex].length) return currQuestions;
+        const resultQuestions = [...(currQuestions || [])];
+        resultQuestions[roundIndex] = [QUESTION_PRESET];
+        return resultQuestions;
+      });
+    }
+  };
+
+  const getHandleQuestionChange = (questionIndex: number) => {
+    return (question: CreatorModeQuestion) => {
```

Comment:
here you can use just setQuestion without returning another function and on line 110 do
```js
onQuestionChange={() => getHandleQuestionChange(i)}
```

Reply:
oh I see now... 
`onQuestionChange={(question) => getHandleQuestionChange(question, i)`